pkg/resources/warehouse.go

```diff
@@ -197,127 +257,118 @@ func CreateWarehouse(d *schema.ResourceData, meta interface{}) error {
-		return err
+	if v := d.Get("enable_query_acceleration").(string); v != "unknown" {
+		parsed, err := strconv.ParseBool(v)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		createOptions.EnableQueryAcceleration = sdk.Bool(parsed)
 	}
-	if err = d.Set("auto_suspend", w.AutoSuspend); err != nil {
-		return err
+	if v := d.Get("query_acceleration_max_scale_factor").(int); v != -1 {
+		createOptions.QueryAccelerationMaxScaleFactor = sdk.Int(v)
 	}
-	if err = d.Set("auto_resume", w.AutoResume); err != nil {
-		return err
+	if v := d.Get("max_concurrency_level").(int); v != -1 {
+		createOptions.MaxConcurrencyLevel = sdk.Int(v)
 	}
-	if err = d.Set("resource_monitor", w.ResourceMonitor); err != nil {
-		return err
+	if v := d.Get("statement_queued_timeout_in_seconds").(int); v != -1 {
+		createOptions.StatementQueuedTimeoutInSeconds = sdk.Int(v)
 	}
-	if err = d.Set("enable_query_acceleration", w.EnableQueryAcceleration); err != nil {
-		return err
+	if v := d.Get("statement_timeout_in_seconds").(int); v != -1 {
+		createOptions.StatementTimeoutInSeconds = sdk.Int(v)
 	}
 
-	err = readWarehouseObjectProperties(d, id, client, ctx)
+	err := client.Warehouses.Create(ctx, id, createOptions)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
+	d.SetId(helpers.EncodeSnowflakeID(id))
+
+	return GetReadWarehouseFunc(false)(ctx, d, meta)
+}
 
-	if w.EnableQueryAcceleration {
-		if err = d.Set("query_acceleration_max_scale_factor", w.QueryAccelerationMaxScaleFactor); err != nil {
-			return err
+func GetReadWarehouseFunc(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+		w, err := client.Warehouses.ShowByID(ctx, id)
+		if err != nil {
+			return diag.FromErr(err)
 		}
-	}
 
-	return nil
-}
+		warehouseParameters, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
+			In: &sdk.ParametersIn{
+				Warehouse: id,
+			},
+		})
+		if err != nil {
+			return diag.FromErr(err)
+		}
 
-func readWarehouseObjectProperties(d *schema.ResourceData, warehouseId sdk.AccountObjectIdentifier, client *sdk.Client, ctx context.Context) error {
-	statementTimeoutInSecondsParameter, err := client.Parameters.ShowObjectParameter(ctx, "STATEMENT_TIMEOUT_IN_SECONDS", sdk.Object{ObjectType: sdk.ObjectTypeWarehouse, Name: warehouseId})
-	if err != nil {
-		return err
-	}
-	logging.DebugLogger.Printf("[DEBUG] STATEMENT_TIMEOUT_IN_SECONDS parameter was fetched: %v", statementTimeoutInSecondsParameter)
-	if err = d.Set("statement_timeout_in_seconds", sdk.ToInt(statementTimeoutInSecondsParameter.Value)); err != nil {
-		return err
-	}
+		if withExternalChangesMarking {
+			if err = handleExternalChangesToObject(d,
+				showMapping{"type", "warehouse_type", string(w.Type), w.Type, nil},
```

Comment:
Hmm, I don't see a problem with this TBH.

IMagine it is not a struct but a function. There is no named parameters concept for functions in golang and this could easily be a function.

Additionally, we read it in IDEs, so it does show the names for these params.